commands2/src/SelectCommandKey.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <robotpy_build.h>
+
+// this assumes that the __hash__ of the python object isn't going to change
+// once added to the map of the SelectCommand. This is... probably reasonable?
+struct SelectCommandKey {
+
+  SelectCommandKey() = default;
+
+  SelectCommandKey &operator=(const py::handle src) {
+    py::gil_scoped_acquire gil;
+    m_v = py::reinterpret_borrow<py::object>(src);
+    m_hash = m_v.attr("__hash__")().cast<std::size_t>();
```

Comment:
isn't this technically incorrect if a `__hash__` attribute is set on an object? `hash()` would still call the real `__hash__` method, but accessing the attribute would not

I believe pybind11 has a helper for this: https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv44hash6handle